### ISSUE TITLE
Scheduler is not triggered or triggered 2 times randomly with native image

### ISSUE THREAD

**Issue by jmpdev34** [quarkusio/quarkus#17724]
## Describe the bug
Randomly, scheduler is not triggered or triggered 2 times with native image
This behavior is not observed in "jvm mode"

### Expected behavior
Scheduler should trigger 1 time and doesn't miss any trigger event with native image

## To Reproduce

https://github.com/jmpdev34/reproducer-scheduler-native

The reproducer should schedule every second based on a cron expression.
When the scheduler method is fired, a log with the datetime is printed
If the scheduler triggers 2 times, a log is printed : 'Already triggered ..."
1. Build a native image of the reproducer project
2. Start runner
3. Wait for a "Already triggered' log and verify if a second hasn't been triggered

Note : if none "Already triggered" log appears, try to shutdown et re-start the runner and/or start another activity on the computer (IT tests on a project for example)

### Configuration
```properties
# Add your application.properties here, if applicable.
reproducer.scheduler.cron=0/1 * * ? * *
```

### Output of `uname -a` or `ver`
Linux 344910zx2006 4.15.0-142-generic #146-Ubuntu SMP Tue Apr 13 01:11:19 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

### Output of `java -version`
openjdk version "11.0.9" 2020-10-20
OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.9+11)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 11.0.9+11, mixed mode)

### GraalVM version (if different from Java)
graalvm-ce-java11-20.1.0

### Quarkus version or git rev
1.11.5.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)
maven version 3.6.3

**Comment by gsmet** [comment]
Are you sure it's not just the event being slightly delayed and you have 2 events in the same second because the one before was a bit delayed?

That being said, a one second delay looks like a lot.

**Comment by jmpdev34** [comment]
Yes I am sure because in production, we observe the same behavior with
scheduler witch fire every 15 minutes

Best regards

Le lun. 7 juin 2021 à 11:30, Guillaume Smet ***@***.***> a
écrit :

**Comment by mkouba** [comment]
Well, I was able to reproduce the `Already triggered ...` behavior, by slowing down my machine significantly. But it's not that the triggers would be fired at once. If you modify [this line](https://github.com/jmpdev34/reproducer-scheduler-native/blob/master/src/main/java/org/acme/ReproducerScheduler.java#L27) to something like `System.out.println("Triggered at : " + now + " but already triggered : " + maps.get(time));` you'll see that the second trigger is fired > 900ms after the first one:
```
Triggered at : 2021-06-07T16:40:11.004974
Triggered at : 2021-06-07T16:40:11.999259 but already triggered : 2021-06-07T16:40:11.004974 executor-thread-1 4
```
That's far from being ideal of course but at least it's fired "roughly every second".

We will need to analyze the scheduling logic and find out we can do better...

**Comment by mkouba** [comment]
> Yes I am sure because in production, we observe the same behavior with scheduler witch fire every 15 minutes

That's a more serious problem though. I think that I know where the problem is. I'll try to find a fix tomorrow...
